osu.Game.Tests/Visual/Multiplayer/TestSceneTeamVersus.cs

```diff
@@ -118,6 +119,25 @@ public void TestChangeTeamsViaButton()
             AddAssert("user still on team 0", () => (client.Room?.Users.FirstOrDefault()?.MatchState as TeamVersusUserState)?.TeamID == 0);
         }
 
+        [Test]
+        public void TestSettingsUpdatedWhenChangingMatchType()
+        {
+            createRoom(() => new Room
+            {
+                Name = { Value = "Test Room" },
+                Type = { Value = MatchType.TeamVersus }
+            });
+
+            AddUntilStep("match type versus", () => client.APIRoom?.Type.Value == MatchType.HeadToHead);
```

Comment:
This assertion is weird. The room seems like it should be created in HeadToHead mode.

Reply:
I just went ahead and fixed this because I don't see any conceivable way that starting in head to head was not the original intention.

I've also had to specify an initial playlist item to fix headless failures, and I added an until step checking that the "create room" button is enabled to boot to make the cause of failure more obvious there (without it it just hangs waiting for a room join that will never happen).